### PR TITLE
feat(auth): sign in via OIDC (PocketID, Authelia, and other providers)

### PR DESCRIPTION
**Pull request by tenkian4**
## Summary

- Adds OIDC sign-in support to `authn.rs`, configured via `RIVEN_SETTING__OIDC_PROVIDERS` (a JSON array, one entry per provider). Endpoints are resolved through OIDC discovery (`{issuer}/.well-known/openid-configuration`) at startup rather than hardcoded, so any spec-compliant provider works — PocketID, Authelia, Keycloak, Authentik, Zitadel, etc. A provider that fails discovery is logged and omitted from the login page rather than failing the whole instance.
- A sign-in whose email matches an existing account auto-links to it. Every configured provider is trusted for this regardless of what it reports for `email_verified`, since its client credentials were wired up by the operator, not chosen by the person signing in.
- New per-provider `disable_sign_up` flag (default `false`, preserving today's behavior): when `true`, a sign-in from that provider only links to an existing account — it never creates one. For deployments where the provider's own user base is broader than who should have riven access.
- Frontend: the login page now fetches `/auth/oidc-providers` and renders a button per resolved provider, wired to `POST /auth/sign-in/social`.
- README, `.env.example`, and `docs/environment.md` updated accordingly.

This was previously supported by the TS frontend's `better-auth` (see the removed `riven-frontend` service's `OAUTH_PROVIDER_POCKETID_*` env vars) but was never ported to the Rust rewrite — the frontend service comment read "Not yet supported on riven-ts rewrite." This closes that gap.

## Test plan

- [x] `cargo fmt --all --check`, `cargo clippy --workspace --all-targets -- -D warnings`, `cargo test --workspace` all clean
- [x] `docs-check` equivalent (`cargo run --bin gen-docs`) — regenerated, only line-number drift, committed
- [x] `schema-check` equivalent — `schema.graphql` and frontend codegen output unchanged (no GraphQL surface touched)
- [x] Frontend `pnpm run check` / `pnpm run lint` / `pnpm run build` all clean
- [x] `cargo audit` — one pre-existing, unrelated advisory (`rkyv`); `Cargo.lock` untouched by this change
- [x] Deployed locally against a real PocketID instance: discovery, login button rendering, full OAuth round-trip, and account linking (including the `account_not_linked` fix for providers that don't report `email_verified`) all verified live
- [ ] Live verification with a second provider (e.g. Authelia) — not tested against a real Authelia instance, only PocketID

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added optional OIDC sign-in with automatic provider discovery.
  * Login now displays configured providers and supports sign-in through the selected provider.
  * Added account linking and configurable provider names, credentials, scopes, and sign-up controls.
  * Added secure endpoint validation and controls for trusting unverified email addresses.

* **Bug Fixes**
  * Providers that cannot be discovered or fail validation are skipped without preventing server startup.

* **Documentation**
  * Added setup instructions, redirect URI guidance, configuration examples, defaults, security guidance, and troubleshooting details.

### CODE REVIEW

File: crates/riven-api/src/server/authn.rs [hunk 1]
```diff
@@ -80,14 +91,32 @@ pub async fn build(
...
         .session_update_age(TimeDelta::days(SESSION_UPDATE_AGE_DAYS))
-        .password_min_length(8);
+        .password_min_length(8)
+        .account(AccountConfig {
+            account_linking: AccountLinkingConfig {
+                trusted_providers: trusted_oidc_providers,
+                ..Default::default()
+            },
+            ..Default::default()
+        });
```
[the remark this answers is not in the document]
Comment (tenkian4) [reply crates/riven-api/src/server/authn.rs]: Fixed in 0579231 — replaced the blanket `trusted_providers` population with a new per-provider `trust_unverified_email` opt-in (default `false`), extracted into a small testable `trusted_oidc_provider_ids` helper with unit tests covering both the default-excluded and opted-in cases. The doc comment on the field spells out the account-takeover risk and when it's actually safe to enable. My own deployment now sets it explicitly for PocketID, since that's the provider that needs it (no email confirmation flow).

File: crates/riven-core/src/settings/oidc.rs [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    /// linked account's `provider_id` — changing it after users have signed
+    /// in orphans their existing links.
+    pub id: String,
+    /// Shown on the login button, e.g. "PocketID". Falls back to `id` when
+    /// empty.
+    #[serde(default)]
+    pub name: String,
+    /// The provider's issuer origin, e.g. `https://pocketid.example.com`
+    /// (no trailing slash or path).
+    pub issuer: String,
```
[the remark this answers is not in the document]
Comment (tenkian4) [reply crates/riven-core/src/settings/oidc.rs]: Fixed in 0579231 — updated the doc comment on `issuer` (and the mirrored README table) to state it must exactly match the provider's advertised issuer and may include a path, with a Keycloak realm example. Confirmed the discovery code already handled this correctly via `trim_end_matches('/')` + append — it was purely a documentation bug.

File: crates/riven-core/src/settings/oidc.rs [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+    pub client_id: String,
+    pub client_secret: String,
+    /// Defaults to `["openid", "profile", "email"]` when empty — the claims
+    /// riven actually reads (`sub`, `email`, `name`, `picture`,
+    /// `email_verified`) all come from those three.
+    #[serde(default)]
+    pub scopes: Vec<String>,
+    /// When `true`, a sign-in from this provider only succeeds if it matches
+    /// an *existing* riven account (by email) — it never creates one.
```
[the remark this answers is not in the document]
Comment (tenkian4) [reply crates/riven-core/src/settings/oidc.rs:36]: Fixed in 0579231 — `discover()` now rejects a discovery document whose `authorization_endpoint`, `token_endpoint` or `userinfo_endpoint` isn't `https://`, with a loopback exception (127.0.0.1/localhost/[::1]) so this module's own tests can keep using a local unencrypted mock server. Added a regression test (`resolve_providers_skips_a_discovery_document_with_a_cleartext_endpoint`) plus direct unit tests for the new `is_https_or_loopback` helper, including the IPv6-brackets edge case.
